src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1235,8 +1236,18 @@ private async Task HandleBreakModeEvent(ResultEventArgs results, BreakRequest br
                     }
                 }
             }
-            else if (reason == "end-stepping-range" || reason == "function-finished")
+            // step over/into
+            else if (reason == "end-stepping-range")
+                _callback.OnStepComplete(thread);
```

Comment:
Also would leave a comment here saying waiting for support for return value in GDB for step over and into. See https://sourceware.org/bugzilla/show_bug.cgi?id=26354